File: wandlet/src/main/java/org/wandledi/wandlet/Page.java
package org.wandledi.wandlet;

import org.wandledi.SelectableImpl;
import org.wandledi.Scroll;

public class Page extends SelectableImpl implements Response {
    private String file;
    
    public Page(String file) {
        super(new Scroll());
        this.file = file;
    }
    
    public String getFile() {
        return file;
    }
}

File: scala-lib/src/main/scala/org/wandledi/wandlet/scala/Page.scala
package org.wandledi.wandlet.scala

import org.wandledi.Scroll
import org.wandledi.wandlet.Response
import org.wandledi.scala.SelectableImpl
import org.wandledi.Selector
import org.wandledi.PathSelector

class Page(val file: String) extends SelectableImpl(new Scroll) with Response {
  override def getFile = file

  /**
   * Extracts the target Element.
   *
   * @param sel Target Element to extract.
   * @param block Block to execute in the context of the extracted Element.
   */
  def extract(sel: Selector)(block: => Unit = {}) {
    get(new PathSelector).extract(sel)
    using(this at sel)(block)
  }

  override def toString = "scala.Page(" + getFile + ")"
}
